### DISCUSSION TITLE
Huge memory usage on restart

### DISCUSSION THREAD

**Discussion by tahayk** [socketio/socket.io#4330] [Q&A] [answered]
Hello, 

I have been using Socket.io in a docker container for a while, and it is running really smooth with almost 8K connexions,
but unfortunately when I restart the container, I notice a huge memory leak,
it jumps from 1-2Gb to 20Gb.

<img width="1339" alt="image" src="https://user-images.githubusercontent.com/22003449/161391978-f4bf66ab-d832-4de8-8987-cfd89b34240e.png">

any idea how to handle this kind of situations ? 

by the way, I'm setting the `maxHttpBufferSize` to 5Mb.

Thank you in advance.

**Comment by darrachequesne** [comment]
Interesting! Have you found anything in the heapdump?

**Reply by tahayk** [reply]
@darrachequesne unfortunately I didn't have access to the heapdump because this happened in our production pod

**Comment by tahayk** [comment, marked as answer]
It turns out that this happened because of a stupid mistake in one of our functions called after each connexion (heavy usage of IO).

**Reply by darrachequesne** [reply]
Good news! Thanks for the update :+1:
